src/builder/conditional.rs

```diff
@@ -266,4 +266,28 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn test_not_all_cases() -> Result<(), BuildError> {
+        let predicate_inputs = vec![type_row![]; 2];
+        let mut builder = ConditionalBuilder::new(predicate_inputs, type_row![], type_row![])?;
+        n_identity(builder.case_builder(0)?)?;
+        match builder.finish_sub_container() {
+            Err(BuildError::ConditionalError(ConditionalBuildError::NotAllCasesBuilt {
+                ..
+            })) => Ok(()),
+            _ => panic!(),
+        }
```

Comment:
you can use `assert_matches!` from `cool_asserts`

Reply:
I tried, but `assert_matches` requires that the `BuildHandle` implements Debug.
I could just call `map(|_| ())` on the success case though...